src/base/bittorrent/session.cpp

```diff
@@ -4035,48 +4077,134 @@ void Session::handleUrlSeedAlert(const lt::url_seed_alert *p)
 
 void Session::handleListenSucceededAlert(const lt::listen_succeeded_alert *p)
 {
-    boost::system::error_code ec;
-    QString proto = "TCP";
-    if (p->sock_type == lt::listen_succeeded_alert::udp)
+    QString proto = "INVALID";
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    switch (p->sock_type)
+    {
+    case lt::listen_succeeded_alert::udp:
+        proto = "UDP";
+        break;
+    case lt::listen_succeeded_alert::tcp:
+        proto = "TCP";
+        break;
+    case lt::listen_succeeded_alert::tcp_ssl:
+        proto = "TCP_SSL";
+        break;
+    case lt::listen_succeeded_alert::i2p:
+        proto = "I2P";
+        break;
+    case lt::listen_succeeded_alert::socks5:
+        proto = "SOCKS5";
+        break;
+    case lt::listen_succeeded_alert::utp_ssl:
+        proto = "UTP_SSL";
+        break;
+    }
+#else
+    switch (p->socket_type)
+    {
+    case lt::socket_type_t::udp:
         proto = "UDP";
-    else if (p->sock_type == lt::listen_succeeded_alert::tcp)
+        break;
+    case lt::socket_type_t::tcp:
         proto = "TCP";
-    else if (p->sock_type == lt::listen_succeeded_alert::tcp_ssl)
+        break;
+    case lt::socket_type_t::tcp_ssl:
         proto = "TCP_SSL";
-    qDebug() << "Successfully listening on " << proto << p->endpoint.address().to_string(ec).c_str() << '/' << p->endpoint.port();
-    Logger::instance()->addMessage(
+        break;
+    case lt::socket_type_t::i2p:
+        proto = "I2P";
+        break;
+    case lt::socket_type_t::socks5:
+        proto = "SOCKS5";
+        break;
+    case lt::socket_type_t::utp_ssl:
+        proto = "UTP_SSL";
+        break;
+    }
+#endif
+
+    boost::system::error_code ec;
+#if (LIBTORRENT_VERSION_NUM < 10200)
+    LogMsg(
         tr("qBittorrent is successfully listening on interface %1 port: %2/%3", "e.g: qBittorrent is successfully listening on interface 192.168.0.1 port: TCP/6881")
             .arg(p->endpoint.address().to_string(ec).c_str(), proto, QString::number(p->endpoint.port())), Log::INFO);
+#else
+    LogMsg(
+        tr("qBittorrent is successfully listening on interface %1 port: %2/%3", "e.g: qBittorrent is successfully listening on interface 192.168.0.1 port: TCP/6881")
+            .arg(p->address.to_string(ec).c_str(), proto, QString::number(p->port)), Log::INFO);
+#endif
 
```

Comment:
You could write it as:
```c++
 LogMsg(
        tr("qBittorrent is successfully listening on interface %1 port: %2/%3", "e.g: qBittorrent is successfully listening on interface 192.168.0.1 port: TCP/6881")
#if (LIBTORRENT_VERSION_NUM < 10200)
            .arg(p->endpoint.address().to_string(ec).c_str(), proto, QString::number(p->endpoint.port())), Log::INFO);
#else
            .arg(p->address.to_string(ec).c_str(), proto, QString::number(p->port)), Log::INFO);
#endif
```

It'll probably help the .ts files and translation efforts.

Reply:
Done.